src/protocol/boolean/bitwise_gt_constant.rs

```diff
@@ -0,0 +1,249 @@
+/// 2.1 Comparison with Bitwise Shared Input – `LTBits` Protocol
+/// Eleftheria Makri, et al.
+/// <https://eprint.iacr.org/2021/119.pdf>
+///
+/// ## Errors
+/// Lots of things may go wrong here, from timeouts to bad output. They will be signalled
+/// back via the error response
+pub async fn bitwise_greater_than_constant<F, C, S>(
+    ctx: C,
+    record_id: RecordId,
+    a: &[S],
+    c: u32,
+) -> Result<S, Error>
+where
+    F: Field,
+    C: Context<F, Share = S>,
+    S: ArithmeticSecretSharing<F>,
+{
+    let first_diff_bit = first_differing_bit(&ctx, record_id, a, c).await?;
+
+    // Compute the dot-product [a] x `first_diff_bit`. 1 iff a > c.
+    // We can swap `a` with `c` to yield 1 iff a < c. We just need to convert
+    // `c` to `&[c]` using `local_secret_shared_bits`.
+    ctx.narrow(&Step::DotProduct)
+        .sum_of_products(record_id, &first_diff_bit, a)
+        .await
+}
+
+async fn first_differing_bit<F, C, S>(
+    ctx: &C,
+    record_id: RecordId,
+    a: &[S],
+    b: u32,
+) -> Result<Vec<S>, Error>
+where
+    F: Field,
+    C: Context<F, Share = S>,
+    S: ArithmeticSecretSharing<F>,
+{
+    let one = ctx.share_of_one();
+
+    // Compute `[a] ^ b`. This step gives us the bits of values where they differ.
+    let mut xored_bits = a
+        .iter()
+        .enumerate()
+        .map(|(i, a_bit)| {
+            // Local XOR
+            if ((b >> i) & 1) == 0 {
+                a_bit.clone()
+            } else {
+                one.clone() - a_bit
+            }
+        })
+        .collect::<Vec<_>>();
+
+    // In the next step, we'll compute prefix-or from MSB to LSB. Reverse the bits order.
+    xored_bits.reverse();
+
+    // Compute prefix-or of the xor'ed bits. This yields 0's followed by 1's with the transition
+    // from 0 to 1 occurring at the index of the first different bit.
+    let prefix_or_context = ctx.narrow(&Step::PrefixOr);
+    let mut prefix_or = Vec::with_capacity(xored_bits.len());
+    prefix_or.push(xored_bits[0].clone());
+    for i in 1..xored_bits.len() {
+        let result = or(
+            prefix_or_context.narrow(&BitOpStep::from(i)),
+            record_id,
+            &prefix_or[i - 1],
+            &xored_bits[i],
+        )
+        .await?;
+        prefix_or.push(result);
+    }
+    // Change the order back to the little-endian format.
+    prefix_or.reverse();
+
+    // Subtract neighboring bits to yield all 0's and a single 1 at the index of the first
+    // differing bit. Note that at the index where the transition from 0 to 1 happens,
+    // `prefix_or[i + 1] > prefix_or[i]`. Do not change the order of the subtraction unless
+    // we use Fp2, or the result will be `[p-1]`.
+    let mut first_diff_bit = Vec::with_capacity(prefix_or.len());
+    for i in 0..(prefix_or.len() - 1) {
+        first_diff_bit.push(prefix_or[i].clone() - &prefix_or[i + 1]);
+    }
+    first_diff_bit.push(prefix_or[prefix_or.len() - 1].clone());
```

Comment:
For a follow-up diff...

...I think we can compute the diff of each bit with its neighbor as a part of the previous loop. In that fashion I think we can avoid some extra allocations. Let's do that in a later PR.